buffer/src/main/java/io/netty/buffer/PoolSubpage.java

```diff
@@ -72,7 +72,10 @@ void init(int elemSize) {
             }
         }
 
-        addToPool();
+        PoolSubpage<T> head = chunk.arena.findSubpagePoolHead(elemSize);
+        synchronized (head) {
+            addToPool(head);
+        }
```

Comment:
I did not do this as we need to hold the lock for `addToPool` and `removeFromPool` here:
https://github.com/netty/netty/blob/synchronize_head/buffer/src/main/java/io/netty/buffer/PoolSubpage.java#L131

So there would be a bit of an overhead as the JVM would need to check that the the lock is already hold when enter addToPool. WDYT ?